File: klink-rest/src/main/java/com/drbrosdev/klinkrest/repository/KlinkRepository.java
package com.drbrosdev.klinkrest.repository;

import com.drbrosdev.klinkrest.model.Klink;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.UUID;

@Repository
public interface KlinkRepository extends JpaRepository<Klink, UUID> {
}

File: klink-realtime/src/main/kotlin/com/example/domain/repository/KlinkRepository.kt
package com.example.domain.repository

import com.example.domain.model.KlinkEntry
import com.example.domain.model.KlinkKeys
import java.util.*

interface KlinkRepository {

    suspend fun insertKlinkEntry(klinkId: UUID, value: String)

    suspend fun deleteKlinkEntryByValue(klinkId: UUID, value: String)

    suspend fun findKeysByKlinkId(klinkId: UUID): KlinkKeys?

    suspend fun insertAll(klinkId: UUID, entries: List<KlinkEntry>)

    suspend fun deleteAllEntriesByKlinkId(klinkId: UUID)

    suspend fun replaceEntries(klinkId: UUID, entries: List<KlinkEntry>)

    suspend fun findEntriesByKlinkId(klinkId: UUID): List<KlinkEntry>
}